### PR TITLE
br: redact ak/sk in logging

### LINKED ISSUE #55273: log-backup: secrets may be directly logging to log when `--send-credentials-to-tikv=false` not set during starting log backup
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. Start a log backup task:
```console
$ tiup br:v8.1.0 log start -u upd-1:2379 --task-name fiolvit -s (s3altpath test)
Starting component `br`: /root/.tiup/components/br/v8.1.0/br log start -u upd-1:2379 --task-name fiolvit -s s3://astro/test?endpoint=http://minio:9000
Detail BR log in /tmp/br.log.2024-08-07T17.45.19+0800
[2024/08/07 17:45:19.859 +08:00] [INFO] [collector.go:77] ["log start"] [streamTaskInfo="{taskName=fiolvit,startTs=451680354413576193,endTS=999999999999999999,tableFilter=*.*}"] [pausing=false] [rangeCount=2]
[2024/08/07 17:45:23.269 +08:00] [INFO] [collector.go:77] ["log start success summary"] [total-ranges=0] [ranges-succeed=0] [ranges-failed=0] [backup-checksum=32.966226ms] [total-take=3.715794709s]
```

And then, check the TiDB log.

### 2. What did you expect to see? (Required)
It shouldn't contain sensetive information.

### 3. What did you see instead (Required)
```
[2024/08/07 17:45:19.844 +08:00] [INFO] [advancer.go:399] ["added event"] [task="storage:<s3:<endpoint:\"http://minio:9000\" region:\"us-east-1\" bucket:\"astro\" prefix:\"test\" access_key:\"minioadmin\" secret_access_key:\"minioadmin\" force_path_style:true > > start_ts:451680354413576193 end_ts:999999999999999999 name:\"fiolvit\" table_filter:\"*.*\" compression_type:ZSTD "] [ranges="{[6D, 6E), [74, 75)}"] [current-checkpoint=451680188790996992]
```

Notice here, our secret key was printed:

```
... access_key:\"minioadmin\" secret_access_key:\"minioadmin\" ...
```

### 4. What is your TiDB version? (Required)
Current master.

**Note**: It is always unsafe to enable `--send-credentials-to-tikv` when starting log backup because: it will store the credentials to PD, and won't rotate them. Then, when the session key expired, there is no way to refresh them(Also anyone that can access PD can query them...). Authorize by IAM roles or other context of the TiKV node are more recommended in productive environment.

### PR DESCRIPTION
**Pull request by RidRisR**
### What problem does this PR solve?

Issue Number: close #55273 

Problem Summary:
log-backup: secrets may be directly logging to log when --send-credentials-to-tikv=false not set during starting log backup

### What changed and how does it work?

Added a wrap of the storage info and redact ak/sk, now the logging woud not print them directly.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

截屏2024-08-23 17 40 01

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Added a wrap of the storage info and redact ak/sk, now the logging woud not print them directly.

```

### CODE REVIEW

File: br/pkg/streamhelper/advancer_cliext.go [hunk 1]
```diff
@@ -64,6 +72,24 @@ func (t *TaskEvent) String() string {
 	return fmt.Sprintf("%s(%s)", t.Type, t.Name)
 }
 
+type TaskInfoRedacted struct {
```
Comment (BornChanger) [inline br/pkg/streamhelper/advancer_cliext.go:75]: please make the masking function to a `pkg/util/redact/redact.go`

File: br/pkg/streamhelper/advancer_cliext.go [hunk 2]
```diff
@@ -23,6 +24,13 @@ import (
 	"go.uber.org/zap"
 )
 
+var (
+	reAccessKey       = regexp.MustCompile(`access_key:\"[^\"]*\"`)
+	reSecretAccessKey = regexp.MustCompile(`secret_access_key:\"[^\"]*\"`)
+	reSharedKey       = regexp.MustCompile(`shared_key:\"[^\"]*\"`)
```
Comment (Leavrth) [inline br/pkg/streamhelper/advancer_cliext.go:30]: Azure blob storage has another credential -- `account-name` and `sas-token`. The `sas-token` is also need to be redacted.

**Comment by YuJuncen** [comment]
```
compilepkg: nogo: errors found by nogo during build-time code analysis:

pkg/util/redact/redact.go:240:7: unused-receiver: method receiver 't' is not referenced in method's body, consider removing or renaming it as _ (revive)

pkg/util/redact/redact.go:236:6: exported: exported type TaskInfoRedacted should have comment or be unexported (revive)
```

Would you fix those linter problems?

**Review by Leavrth — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**